src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -304,6 +304,14 @@ internal virtual ImportChain ImportChain
             }
         }
 
+        internal virtual HashSet<string> QuickTypeIdentifierAttributeCheckSet
```

Comment:
💡 Consider using `ImmutableHashSet<string>`. Since the algorithm allows reference sharing between certain binders, immutability helps reason about the code. In addition, the majority case involves only a single element in the set, and `ImmutableHashSet<string>` should perform similarly to `HashSet<string>` for this case.

:memo: This is a "strong suggestion"; it appears to be a clear win for maintainability, and should *at least* be faster than the original implementation which had generally-acceptable performance.